ams/ams-server/src/main/java/com/netease/arctic/ams/server/utils/ChangeFilesUtil.java

```diff
@@ -72,4 +73,114 @@ private static void deleteChangeFiles(KeyedTable keyedTable, List<DataFile> chan
     changeFiles.forEach(changeDelete::deleteFile);
     changeDelete.commit();
   }
+
+  /**
+   * Select all the files whose sequence <= maxSequence as Selected-Files, seek the maxSequence to find as many
+   * Selected-Files as possible, and also
+   * - the cnt of these Selected-Files must <= maxFileCntLimit
+   * - the max TransactionId of the Selected-Files must > the min TransactionId of all the left files
+   *
+   * @param snapshotFileGroups snapshotFileGroups
+   * @param maxFileCntLimit    maxFileCntLimit
+   * @return the max sequence of selected file, return Long.MAX_VALUE if all files should be selected,
+   * Long.MIN_VALUE means no files should be selected
+   */
+  public static long getMaxSequenceLimit(List<SnapshotFileGroup> snapshotFileGroups, long maxFileCntLimit) {
```

Comment:
I don't think this is a good util method as it is hard to reuse.
Maybe we can abstract the change store file planning process to a util method like `planOptimizingFiles`.

Reply:
This method is not a good abstraction. I will refactor it.

Reply:
I moved this method from `ChangeFilesUtils` to `TableOptimizeItem`, corresponding to scan base files.